src/mappings/rewardsDistributor.ts

```diff
@@ -0,0 +1,34 @@
+import { AcalaEvmEvent } from "@subql/acala-evm-processor";
+import { ClaimRewardsEvent } from "@subql/contract-sdk/typechain/RewardsDistributer";
+import assert from "assert";
+import { Trader } from "../types";
+import { getUpsertAt } from "./utils";
+
+export async function handleRewardsClaimed(
+    event: AcalaEvmEvent<ClaimRewardsEvent['args']>
+  ): Promise<void> {
+    logger.info('handleRewardsClaimed');
+    assert(event.args, 'No event args');
+
+    const { delegator, rewards } = event.args;
+    const rewardsBigInt = rewards.toBigInt();
+    const handlerInfo = getUpsertAt('handleRewardsClaimed', event);
+    
+    let trader = await Trader.get(delegator);
+
+    if (!trader) {
```

Comment:
I think `if (!trader)` we don't have to handle for this part, if the user don't swap then there is no reason to create a trader record

Reply:
We need to calculate `totalAwardAmount` even though user don't swap after claim reward